flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/KinesisConfigUtil.java

```diff
@@ -191,6 +191,7 @@ public static KinesisProducerConfiguration getValidatedProducerConfiguration(Pro
 		}
 
 		KinesisProducerConfiguration kpc = KinesisProducerConfiguration.fromProperties(config);
+		kpc.setRegion(config.getProperty(AWSConfigConstants.AWS_REGION));
```

Comment:
Does this work if the region property is `null`?

Reply:
I don't think so, but it is already safeguarded with https://github.com/apache/flink/pull/5160/files/72be9da2048d4e2cb6c855ce16c9652556e1f762#diff-16d72c003634fdee6e6efa5ece15c385R186 that the region property is not null.